packages/api-client/src/components/AddressBar/AddressBarServerItem.vue

```diff
@@ -38,28 +38,91 @@ const updateSelectedServer = (serverUid: string) => {
 /** Set server checkbox in the dropdown */
 const isSelectedServer = (serverId: string) =>
   activeServer.value?.uid === serverId
+
+const serverVariables = computed(() => {
+  const server = activeServer.value
+  if (!server) return []
+
+  return Object.entries(server.variables || {}).map(([key, variable]) => ({
+    key,
+    value: variable.default || '',
+  }))
+})
+
+/** Prevents menu from closing if the server has variables */
+const handleClick = (event: Event, serverId: string) => {
+  const hasVariables =
+    servers[serverId]?.variables &&
+    Object.keys(servers[serverId].variables).length > 0
+
+  if (hasVariables) {
+    event.stopPropagation()
+  }
+  updateSelectedServer(serverId)
+}
+
+const updateServerVariable = (key: string, value: string) => {
+  if (!activeServer.value) return
+
+  const variables = activeServer.value.variables || {}
+  variables[key] = { ...variables[key], default: value }
+
+  serverMutators.edit(activeServer.value.uid, 'variables', variables)
+}
```

Comment:
We are setting the value as default here, would it make more sense to add a value property? Because now we can never go back to the default. Or is that ok?

Reply:
i removed this, it was initially to test out in my first iteration variables presence.